File: UAUTitanium/UAUMobileWebAPI/UAUMobileWebAPI/Models/ProcessoPagamento/ProcessoDePagamento.vb
Namespace Models.ProcessoPagamento
    Public Class ProcessoDePagamento
        Public Property Row As Integer
        Public Property DocumentoFiscal As String
        Public Property ValorDocumentoFiscal As Double
        Public Property Acrescimo As Double
        Public Property Desconto As Double
        Public Property NumeroCotacao As Integer
        Public Property OrdemCompra As Integer
        Public Property Contrato As Integer
        Public Property Medicao As Integer
        Public Property NumeroProcesso As String
        Public Property EmpresaObra As String
        Public Property Fornecedor As String
        Public Property DataProrrogacao() As Object
            Get
                Return dataprorrogacao_value
            End Get
            Set(value As Object)
                Try
                    If Not IsDBNull(value) Then
                        Dim dt = CType(value, DateTime)
                        dataprorrogacao_value = dt.ToString("yyyy-MM-dd HH:mm:ss")
                    End If
                Catch ex As Exception
                    Throw
                End Try
            End Set
        End Property
        Private dataprorrogacao_value As String
        Public Property Parcela As Integer
        Public Property Aprovado As Boolean
        Public Property ValPagar As Double
        Public Property ValorProcesso As Double
        Public Property TipoEmissao As String
        Public Property TipoProcesso As Integer
        Public Property DataVencimento() As Object
            Get
                Return datavencimento_value
            End Get
            Set(value As Object)
                Try
                    If Not IsDBNull(value) Then
                        Dim dt = CType(value, DateTime)
                        datavencimento_value = dt.ToString("yyyy-MM-dd HH:mm:ss")
                    End If
                Catch ex As Exception
                    Throw
                End Try
            End Set
        End Property
        Private datavencimento_value As String
        Public Property DataGeracao() As Object
            Get
                Return datageracao_value
            End Get
            Set(value As Object)
                Try
                    If Not IsDBNull(value) Then
                        Dim dt = CType(value, DateTime)
                        datageracao_value = dt.ToString("yyyy-MM-dd HH:mm:ss")
                    End If
                Catch ex As Exception
                    Throw
                End Try
            End Set
        End Property
        Private datageracao_value As String
        Public Property BancoConta As String
        Public Property Empresa As String
        Public Property EmpresaCod As Integer
        Public Property Obra As String
        Public Property ObraCod As String
        Public Property Observacao() As Object
            Get
                Return obs_value
            End Get
            Set(value As Object)
                Try
                    If Not IsDBNull(value) Then
                        obs_value = value
                    Else
                        obs_value = ""
                    End If
                Catch ex As Exception
                    Throw
                End Try
            End Set
        End Property
        Private obs_value As String
    End Class
End Namespace

File: UAUTitanium/UAU Coorporativo/app/models/ProcessoDePagamento.js
/** @class Models.ProcessoDePagamento
 * Modelo que representa as informações do processo de pagamento.
 * @private
 * @alteracao 04/03/2015 180419 Projeto Carlos Eduardo Santos Alves Domingos
 * Criação.
 * @alteracao 10/04/2015 185753 Projeto Carlos Eduardo Santos Alves Domingos
 * Adicionada a coluna ValPagar.
 * @cfg {Number} NumeroProcesso Número da tarefa.
 * @cfg {String} EmpresaObra Descrição da novidade.
 * @cfg {String} Fornecedor Link para o virtuau.
 * @cfg {Date} DataProrrogacao Número da tarefa.
 * @cfg {Number} Parcela Descrição da novidade.
 * @cfg {Boolean} Aprovado Link para o virtuau.
 * @cfg {Number} ValorProcesso Número da tarefa.
 * @cfg {String} TipoEmissao Descrição da novidade.
 * @cfg {Number} TipoProcesso Link para o virtuau.
 * @cfg {Date} DataVencimento Número da tarefa.
 * @cfg {Date} DataGeracao Descrição da novidade.
 * @cfg {String} BancoConta Link para o virtuau.
 * @cfg {String} Empresa Número da tarefa.
 * @cfg {Number} EmpresaCod Descrição da novidade.
 * @cfg {String} ObraCod Link para o virtuau.
 * @cfg {String} Obra Número da tarefa.
 * @cfg {String} Observacao Descrição da novidade.
 * @cfg {String} DocumentoFiscal Descrição do documento fiscal vinculado.
 * @cfg {Number} ValorDocumentoFiscal Valor do documento fiscal vinculado.
 * @cfg {Number} Acrescimo Acréscimo do processo.
 * @cfg {Number} Desconto Desconto do processo.
 * @cfg {Number} NumeroCotacao Número da cotação.
 * @cfg {Number} OrdemCompra Ordem de compra.
 * @cfg {Number} Contrato Contrato.
 * @cfg {Number} Medicao Medição.
 * @cfg {Number} ValPagar Valor liquido do processo: liquido = valor - desconto + acrescimo.
 */
exports.definition = {
	config: {
		columns: { 
			"NumeroProcesso": "integer",
		    "EmpresaObra": "string",
		    "Fornecedor": "string",
		    "DataProrrogacao": "datetime",
		    "Parcela": "integer",
		    "Aprovado": "boolean",
		    "ValorProcesso": "double",
		    "ValPagar": "double",
		    "TipoEmissao": "string",
		    "TipoProcesso": "integer",
		    "DataVencimento": "datetime",
		    "DataGeracao": "datetime",
		    "BancoConta": "string",
		    "Empresa": "string",
		    "EmpresaCod": "integer",
		    "ObraCod": "string",
		    "Obra": "string",
		    "Observacao": "string",
		    "DocumentoFiscal" : "String",
	        "ValorDocumentoFiscal" : "Double",
	        "Acrescimo" : "Double",
	        "Desconto" : "Double",
	        "NumeroCotacao" : "Integer",
	        "OrdemCompra" : "Integer",
	        "Contrato" : "Integer",
	        "Medicao" : "Integer"
		},
		adapter: {
			type: "properties",
			collection_name: "ProcessoDePagamento"
		}
	},
	extendModel: function(Model) {
		_.extend(Model.prototype, {
			// extended functions and properties go here
		});

		return Model;
	},
	extendCollection: function(Collection) {
		_.extend(Collection.prototype, {
			// extended functions and properties go here
			comparator: function(model){
				return Alloy.Globals.format.FormatoAnoMesDia(model.get("DataProrrogacao"));
			}
		});

		return Collection;
	}
};